engine/src/main/java/org/terasology/rendering/nui/editor/layers/AbstractEditorScreen.java

```diff
@@ -113,6 +118,15 @@
     protected abstract Path getAutosaveFile();
 
     @Override
+    public void onDraw(Canvas canvas) {
+        super.onDraw(canvas);
+        if (!autosaveLoaded) {
+            loadAutosave();
+            autosaveLoaded = true;
+        }
```

Comment:
That looks like a hack to me (no one would expect a autosave loading in a draw method). A comment why it is done there should at least be added.